App/Containers/Root/RootScreen.js

```diff
@@ -1,37 +1,57 @@
 import React, { Component } from 'react'
 import NavigationService from 'App/Services/NavigationService'
 import AppNavigator from 'App/Navigators/AppNavigator'
-import { View } from 'react-native'
+import { View, Text } from 'react-native'
 import { connect } from 'react-redux'
 import StartupActions from 'App/Stores/Startup/Actions'
 import { PropTypes } from 'prop-types'
 import { Helpers } from 'App/Theme'
+import LoginScreen from '../LoginScreen/LoginScreen'
 
 class RootScreen extends Component {
   componentDidMount() {
-    // Run the startup saga when the application is starting
-    this.props.startup()
+    // Run the startup saga only if userId is available
+    if (this.props.userId !== null)
+       this.props.startup()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.userId !== prevProps.userId) {
```

Comment:
add another condition `&& this.props.userId` as if userId changed to null/undefinded somehow the app will enter startUp saga again